src/pal/src/locale/locale.cpp

```diff
@@ -359,7 +359,7 @@ static CFStringRef CFStringCreateMacFormattedLocaleName(LPCWSTR lpLocaleName)
         return NULL;
     }
 
-    CFStringAppendCharacters(cfMutableLocaleName, lpLocaleName, PAL_wcslen(lpLocaleName));
+    CFStringAppendCharacters(cfMutableLocaleName, (const UniChar*)lpLocaleName, PAL_wcslen(lpLocaleName));
```

Comment:
Not sure I follow, you mean something like this:

https://gist.github.com/kangaroo/95ba61cf32338a4d5368